vms/platformvm/state/disk_staker_diff_iterator.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/hashing"
+)
+
+const (
+	startWeightKeyLength  = hashing.HashLen + database.Uint64Size
+	weightKeyLength       = startWeightKeyLength + hashing.AddrLen
+	weightKeyHeightOffset = hashing.HashLen
+	weightKeyNodeIDOffset = weightKeyHeightOffset + database.Uint64Size
+
+	weightValueLength = 1 + database.Uint64Size
+	weightValueTrue   = 0x01
+
+	startBLSKeyLength  = database.Uint64Size
+	blsKeyLength       = startBLSKeyLength + hashing.AddrLen
+	blsKeyNodeIDOffset = database.Uint64Size
+)
+
+var (
+	errUnexpectedWeightKeyLength   = fmt.Errorf("expected weight key length %d", weightKeyLength)
+	errUnexpectedWeightValueLength = fmt.Errorf("expected weight value length %d", weightValueLength)
+	errUnexpectedBLSKeyLength      = fmt.Errorf("expected bls key length %d", blsKeyLength)
+)
+
+// getStartWeightKey is used to determine the starting key when iterating.
+//
+// Note: the result should be a prefix of [getWeightKey] if called with the same
+// arguments.
+func getStartWeightKey(subnetID ids.ID, height uint64) []byte {
+	key := make([]byte, startWeightKeyLength)
+	copy(key, subnetID[:])
+	// Note: [height] is encoded as a bit flipped big endian number so that
+	// iterating lexicographically results in iterating in decreasing heights.
+	binary.BigEndian.PutUint64(key[weightKeyHeightOffset:], ^height)
```

Comment:
All the bit-flipping seems pretty fragile to track (albeit in one file) - can we extract it into a helper function? maybe `putHeight` and `getHeight`?

Reply:
moved into helpers